matchms/filtering/add_retention.py

```diff
@@ -0,0 +1,83 @@
+from typing import Any
+from typing import List
+from matchms.utils import get_first_common_element
+from ..typing import SpectrumType
+
+
+_retention_time_keys = ["retention_time", "retentiontime", "rt", "scan_start_time"]
+_retention_index_keys = ["retention_index", "retentionindex", "ri"]
+
+
+def safe_store_value(spectrum: SpectrumType, value: Any, target_key: str) -> SpectrumType:
+    """Helper function to safely store a value in the target key without throwing an exception, but storing 'None' instead.
+
+    Args:
+        spectrum (SpectrumType): Spectrum to which to add 'value' in 'target_key'.
+        value (Any): Value to parse into 'target_key'.
+        target_key (str): Name of the key in which to store the value.
+
+    Returns:
+        SpectrumType: Spectrum with added key.
+    """
+    if value is not None:   # one of accepted keys is present
+        try:
+            value = float(value)

```

Comment:
For some weird reason multiple file types I have seen so far had the retention time stored as list of floats like this:
'scan_start_time': [0.627583333333]
Which would not be loaded in the current setup. 
So as suggested change, something like:
if isinstance(value, list):
    if len(value) == 1:
        value = value[0]

Reply:
Please see the changes in https://github.com/matchms/matchms/pull/265/commits/e22ed8f5278a6efe95e1146a3d334d1f7d3dc840